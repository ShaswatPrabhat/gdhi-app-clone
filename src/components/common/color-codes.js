export default {
  getColorCodes() {
    // Update variables if changed.
    return [
      { score: "5", color: "#01C975", description: "Most Developed" },
      { score: "4", color: "#80E1CC", description: "" },
      { score: "3", color: "#FFE180", description: "" },
      { score: "2", color: "#FFCA82", description: "" },
      { score: "1", color: "#FCAB9C", description: "Least Developed" },
      { score: "NA", color: "#6C757D" },
    ];
  },
};
