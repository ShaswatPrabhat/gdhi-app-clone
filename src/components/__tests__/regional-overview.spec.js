import RegionalOverview from "../../components/regionalOverview/regional-overview.vue";
import { describe, it, expect, vi } from "vitest";
import VueRouter from "vue-router";
import { EventBus } from "../common/event-bus";
import { createLocalVue, shallowMount } from "@vue/test-utils";
import { i18n } from "../../plugins/i18n";
import flushPromises from "flush-promises";
describe("Region Overview", () => {
  let wrapper;
  const eventBusOnSpy = vi.spyOn(EventBus, "$on");
  const localVue = createLocalVue();
  localVue.use(VueRouter);
  const router = new VueRouter();
  const regionData = [
    { regionId: "PAHO", regionName: "Pan American Region" },
    { regionId: "AFRO", regionName: "African Region" },
    { regionId: "EURO", regionName: "European Region" },
    { regionId: "WPRO", regionName: "Western Pacific Region" },
    { regionId: "SEARO", regionName: "South-East Asian Region" },
    { regionId: "EMRO", regionName: "Eastern Mediterranean Region" },
  ];
  window.appProperties = {
    getRegions: function () {
      return regionData;
    },
  };

  beforeEach(async () => {
    wrapper = shallowMount(RegionalOverview, {
      localVue,
      router,
      i18n,
    });
    await flushPromises();
  });

  afterEach(() => {
    wrapper.vm.$route.params.regionId = "";
  });

  it("should render regional overview", async () => {
    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it("should have the data", async () => {
    wrapper.vm.$route.params.regionId = "PAHO";
    wrapper.vm.getRegionNameFromRegionId();
    expect(wrapper.vm.regionName).equal("Pan American Regional Overview");
  });

  it("should register event when regional overview page is mounted", async () => {
    expect(eventBusOnSpy.mock.calls[0][0]).to.equal("region:translated");
  });
});
