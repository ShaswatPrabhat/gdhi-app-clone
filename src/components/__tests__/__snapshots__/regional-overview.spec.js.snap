// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Region Overview > should render regional overview 1`] = `
<div
  class="region-container"
  data-v-e090c329=""
>
  <p
    class="heading"
    data-v-e090c329=""
  >
    
  </p>
  <div
    class="region-year-container"
    data-v-e090c329=""
  >
    <region-default-year-stub
      data-v-e090c329=""
    />
    <region-year-selector-stub
      data-v-e090c329=""
    />
    <!---->
  </div>
</div>
`;
